crates/client/src/storage.rs

```diff
@@ -32,33 +33,57 @@ pub trait RegistryStorage: Send + Sync {
     /// Reset registry local data
     async fn reset(&self, all_registries: bool) -> Result<()>;
 
+    // /// Directory where all registries are stored
+    // fn registries_dir(&self) -> PathBuf;
     /// Loads most recent checkpoint
-    async fn load_checkpoint(&self) -> Result<Option<SerdeEnvelope<TimestampedCheckpoint>>>;
+    async fn load_checkpoint(
+        &self,
+        namespace_registry: &Option<HeaderValue>,
```

Comment:
I'd highly recommend wrapping `HeaderValue` in a new type like `RegistryName` (or `RegistryDomain`?); you really don't want your trait interfaces to depend on a specific external library type like this. Something like:

```rust
pub struct RegistryName(String);

impl TryFrom<HeaderValue> for RegistryName ...

impl From<RegistryName> for HeaderValue ...
```